walk-in-interview/src/main/webapp/applicant-interested-list/script.js

```diff
@@ -0,0 +1,193 @@
+    method: 'GET',
+    headers: {'Content-Type': 'application/json'},
+  })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log('data', data);
+        // Resets the error (there might have been an error msg from earlier)
+        setErrorMessage(/* errorMessageElementId= */ 'error-message',
+            /* msg= */ '', /* includesDefaultMsg= */ false);
+        return data;
+      });
+}
+
+/**
+ * This will add all the interested job listings onto the page.
+ * 
+ * @param {Object} jobPageData The details to be shown on the page.
+ */
+function displayInterestedJobListings(jobPageData) {
+  const jobListingsElement = document.getElementById('job-listings');
+  const jobShowing = document.getElementById('job-listings-showing');
+
+  /* reset the list so we don't render the same jobs twice */
+  jobListingsElement.innerHTML = '';
+  jobShowing.innerHTML = '';
+
+  if (jobPageData === undefined ||
+    !jobPageData.hasOwnProperty('jobList') ||
+    jobPageData['jobList'].length === 0) {
+    setErrorMessage(/* errorMessageElementId= */ 'error-message',
+        /* msg= */ STRINGS['no-jobs-error-message'],
+        /* includesDefaultMsg= */ false);
+    return;
+  }
+
+  const jobListings = jobPageData['jobList'];
+
+  jobListings.forEach((job) => {
+    jobListingsElement.appendChild(buildJobElement(job));
+  });
+
+  jobShowing.innerText = `${jobPageData['range'].minimum} -` +
+    ` ${jobPageData['range'].maximum} ${STRINGS['job-listings-showing']} ` +
+    `${jobPageData['totalCount']}`;
+}
+
+/**
+ * Builds the job element given the job details from the servlet response.
+ * 
+ * @param {Object} job The job to be displayed.
+ * @return {Element} The job listing element.
+ */
+function buildJobElement(job) {
+  const jobPostPreviewTemplate =
+    document.getElementById('job-listing-template');
+
+  const jobPostPreview =
+    jobPostPreviewTemplate.cloneNode( /* and child elements */ true);
+  jobPostPreview.setAttribute('id', job['jobId']);
+
+  const jobTitle = jobPostPreview.children[0];
+  jobTitle.innerText = job['jobTitle'];
+
+  const jobAddress = jobPostPreview.children[1];
+  const location = job['jobLocation'];
+  jobAddress.innerText = `${location['address']}, ${location['postalCode']}`;
+
+  const jobPay = jobPostPreview.children[2];
+  const pay = job['jobPay'];
+  jobPay.innerText = `${pay['min']} - ${pay['max']} SGD ` +
+  `(${pay['paymentFrequency'].toLowerCase()})`;
+
+  const requirementsList = jobPostPreview.children[3];
+  const fullRequirementsList = getRequirementsList();
+  const requirementsArr = [];
+
+  const jobRequirements = job['requirements'];
+  for (const key in jobRequirements) {
+    if (jobRequirements.hasOwnProperty(key)) {
+      if (jobRequirements[key] /* is true */) {
+        requirementsArr.push(fullRequirementsList[key]);
+      }
+    }
+  }
+
+  requirementsList.innerText =
+  `Requirements List: ${requirementsArr.join(', ')}`;
```

Comment:
"Requirements List:" should come from Strings not hardcoded. (probably your requirements separator too). Also is requirementsArr already translated by the server (such that it's safe to dump as-is onto the page)

Reply:
requirements are translated when rendering